src/main.cpp

```diff
@@ -1,53 +1,109 @@
+    const auto model_bb = GetModelBoundingBox(hex_models.at(0));
+    const auto model_size = Vector3Subtract(model_bb.max, model_bb.min);
+    const float scale = 2.0f / std::max({model_size.x, model_size.y, model_size.z});
+    
+    // for draging the map around
+    Vector3 mouse_grab_point;
+
+    CylinderHexWorld<char> world (200, 100, (char)0, (char)4);
 
     while(!WindowShouldClose()) {
-        UpdateCamera(&camera);
+        // for some reason, dragging around is unstable
+        // i know, that the logical cursor is slightly delayed, but still, it should be delayed
+        // equally for all of the frame. The exact pointthat is selected will be slightly changing,
+        // i very much doubt it's because of float rounding. for the future to solve
+        const auto mouse_ray = GetMouseRay(GetMousePosition(), camera);
+        const auto mouse_on_ground = intersect_with_ground_plane(mouse_ray, 0.2f);
+        const auto hovered_coords = HexCoords::from_world_unscaled(mouse_on_ground.x, mouse_on_ground.z);
+        if (IsMouseButtonPressed(MOUSE_BUTTON_LEFT)) {
+            mouse_grab_point = mouse_on_ground;
+        }
+        if (IsMouseButtonDown(MOUSE_BUTTON_LEFT)) {
+            const auto grab_on_screen = GetWorldToScreen(mouse_grab_point, camera);
+            const auto grab_in_world = intersect_with_ground_plane(GetMouseRay(grab_on_screen, camera), 0.2f);
+            const auto grab_offset = Vector3Subtract(grab_in_world, mouse_on_ground);
+            camera.position = Vector3Add(camera.position, grab_offset);
+            camera.target = Vector3Add(camera.target, grab_offset);
+        }
+
+        const auto top_left = intersect_with_ground_plane(GetMouseRay(Vector2{0, 0}, camera), 0.0f);
+        const auto top_right = intersect_with_ground_plane(GetMouseRay(Vector2{(float)GetScreenWidth(), 0}, camera), 0.0f);
+        const auto bottom_left = intersect_with_ground_plane(GetMouseRay(Vector2{0, (float)GetScreenHeight()}, camera), 0.0f);
+        const auto bottom_right = intersect_with_ground_plane(GetMouseRay({(float)GetScreenWidth(), (float)GetScreenHeight()}, camera), 0.0f);
+
+        const auto to_render = world.all_within_unscaled_quad(
+            {top_left.x, top_left.z},    
+            {top_right.x, top_right.z},    
+            {bottom_left.x, bottom_left.z},    
+            {bottom_right.x, bottom_right.z}    
+        );
+
+        if (IsMouseButtonPressed(MOUSE_BUTTON_RIGHT)) {
+            if (auto hx = world.at_ref_abnormal(hovered_coords)) {
+                hx.value().get() = (hx.value().get() + 1) % 4;
+            }
+        }
+
+        // this is temporary and also terrible, and also shows the bad frustom in all_within_unscaled_quad
+        const auto scroll = GetMouseWheelMove();
+        if (abs(scroll) > 0.01f) {
+            camera.fovy = Clamp(camera.fovy + scroll * 3.0f, 30.0f, 110.0f);
+        };
 
         BeginDrawing();
             ClearBackground(WHITE);

```

Comment:
Co myślisz o użyciu klamerek do wydzielenia bloków?
{
  ClearBackground(WHITE);
  ...
}
Pozwoli to na autoformatowanie kodu w poprawny sposób